### PR TITLE
Update Barong::App validations

### PR DESCRIPTION
**Pull request by dnfd**

### CODE REVIEW

File: config/initializers/cors.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Barong::App.define do |config|
   config.set(:api_cors_origins, '*')
-  config.set(:api_cors_max_age, '3600')
-  config.set(:api_cors_allow_credentials, 'false')
+  config.set(:api_cors_max_age, '3600', regex: /^\d+$/)
+  config.set(:api_cors_allow_credentials, 'false', values: %w(false true))
```
Comment (mod) [inline config/initializers/cors.rb]: maybe we should convert as a boolean after?

Comment (dnfd) [reply config/initializers/cors.rb]: Yes, I thought about `type: :bool`

**Comment by ec** [comment]
Thats an awesome improvement 
We should finish it as soon as possible)
It gives a lot more clearness and power for config store

@dnfd nice work!

**Comment by ec** [comment]
@mod 
As we have moving so far with this, so maybe its time to prefix all envs with BARONG in the Barong::App class?
The idea is - we ask to provide env with a prefix, bur remove it in config.set

Then we will not have a duplicate in naming, like
`Barong::App.config.#{config_name}` and not `Barong::App.config.barong_#{config_name}`
But DevOps will be forced to set envs with `BARONG_CONFIG_NAME`

It will clarify a lot our deployments.
Although it's a breaking change for the current one (everything has its price)

nice catch from @dnfd

**Review by ec — APPROVED** [review]

**Pull request merged**